index.html

```diff
@@ -414,19 +416,18 @@
             the same flag manually when using the Fetch API. All requests with
             this flag set share the same in-flight quota restrictions that is
             enforced within the Fetch API.</p>
-            <li>Let <var>corsMode</var> be "<code>cors</code>".</li>
-            <li>Let <var>headerList</var> be <code>null</code>.</li>
-          </ul>
-        </li>
-        <li>If <var>mimeType</var> is not null:
-          <ul>
-            <li>If <var>mimeType</var> value is a <a>CORS-safelisted
-            request-header</a> value for the <code>Content-Type</code> header,
-            set <var>corsMode</var> to "<code>no-cors</code>".
+            <li>Set <var>corsMode</var> to "<code>cors</code>".</li>
```

Comment:
is it intentional that you chose to set the mode to cors here and then revert at L424 than setting the mode to cors when the contentType is not CORS safelisted?

this means that the mode will be cors when contentType is null here.

Reply:
Yes, because the intent is to follow a (strict) whitelist model: https://github.com/w3c/beacon/pull/33#issuecomment-229896895.

Reply:
/cc @annevk 

Yeah, it's important that we set the default to the stronger mode. The switch from blacklist to whitelist in https://github.com/w3c/beacon/commit/1073d61c75838dcf1d56c4456f6dc5c791fd5f42 itself really makes sense. But the concern raised by Anne was caused by:
- setting the mode separately and far away from the logic to set the Content-Type without checking what's going to be set to the Content-Type.
- switching on the type here inside the beacon spec.

The current spec doesn't do the switching on the type here but is depending on the fetch spec by invoking the `extract` algorithm. So, the second point is no longer a concern.

The current code is written in the form that determines both of the mode and the Content-Type to append at the same place. With this, I don't think we should be too much careful.

Setting the mode to cors when we're not adding any Content-Type header doesn't make sense and we should fix it if possible. As the concerns are gone as described above, I think we can consider fixing it. WDTY?

I'm fine with initializing the mode to cors at L417, but want to suggest that we add "Otherwise" step between L428 and L429 to set corsMode to no-cors.

Reply:
> Setting the mode to cors when (we have a POST payload and) we're not adding any Content-Type header doesn't make sense and we should fix it if possible.

@toddreifsteck @annevk thoughts?

Reply:
That's fine.